executor/distsql.go

```diff
@@ -525,7 +534,7 @@ func (e *IndexLookUpExecutor) buildTableReader(ctx context.Context, handles []in
 
 // Close implements Exec Close interface.
 func (e *IndexLookUpExecutor) Close() error {
-	if e.finished == nil {
+	if !e.workerStarted || e.finished == nil {
```

Comment:
we cannot distinguish 'workerInit' and `workerStopped` status, so maybe we see `workerStarted==false` but worker be started later

Reply:
Yes, but we don't need to distinguish these two status now so it doesn't matter?